### PR TITLE
bugfix:code optimization of EurekaRegistryServiceImpl

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
1.Remove redundant static keywords
2.initial size of map.
3.lazy load of some object.

### Ⅱ. Does this pull request fix one issue?

#753 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
I've done integration testing.

### CODE REVIEW

File: config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java [hunk 1]
```diff
@@ -61,9 +61,13 @@
 
     public ZookeeperConfiguration() {
         if (zkClient == null) {
-            zkClient = new ZkClient(FILE_CONFIG.getConfig(FILE_CONFIG_KEY_PREFIX + SERVER_ADDR_KEY),
-                FILE_CONFIG.getInt(FILE_CONFIG_KEY_PREFIX + SESSION_TIME_OUT_KEY),
-                FILE_CONFIG.getInt(FILE_CONFIG_KEY_PREFIX + CONNECT_TIME_OUT_KEY));
+            synchronized (ZookeeperConfiguration.class) {
+                if (null == zkClient) {
+                    zkClient = new ZkClient(FILE_CONFIG.getConfig(FILE_CONFIG_KEY_PREFIX + SERVER_ADDR_KEY),
+                        FILE_CONFIG.getInt(FILE_CONFIG_KEY_PREFIX + SESSION_TIME_OUT_KEY),
```
Comment (slievrly) [inline config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java]: SESSION_TIME_OUT and CONNECT_TIME_OUT_KEY add config default value

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java [hunk 2]
```diff
@@ -199,27 +206,25 @@ private static Properties getEurekaProperties(boolean needRegister) {
...
         }
         return application;
     }
 
-    private static EurekaClient getEurekaClient(boolean needRegister) throws EurekaRegistryException{
+    private EurekaClient getEurekaClient(boolean needRegister) throws EurekaRegistryException {
         if (eurekaClient == null) {
             synchronized (EurekaRegistryServiceImpl.class) {
                 try {
                     if (eurekaClient != null) {
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java]: move ，double check singleton

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java [hunk 3]
```diff
@@ -63,23 +64,30 @@
...
+    }
 
     public static EurekaRegistryServiceImpl getInstance() {
         if (null == instance) {
             synchronized (EurekaRegistryServiceImpl.class) {
                 if (null == instance) {
                     instance = new EurekaRegistryServiceImpl();
+                    CLUSTER_ADDRESS_MAP = new ConcurrentHashMap<>(MAP_INITIAL_CAPACITY);
+                    instanceConfig = new CustomEurekaInstanceConfig();
+                    InstanceInfo instanceInfo = new EurekaConfigBasedInstanceInfoProvider(instanceConfig).get();
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java]: The previous logic config is initialized, and then the instance is built. You are constructing an empty instance.

File: config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java (hunk 1 above)
Comment (xingfudeshi) [reply config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java]: ok.

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java (hunk 3 above)
Comment (xingfudeshi) [reply discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java]: my fault.i will fix it.

File: config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java (hunk 1 above)
Comment (xingfudeshi) [reply config/src/main/java/com/alibaba/fescar/config/ZookeeperConfiguration.java]: done

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java (hunk 3 above)
Comment (xingfudeshi) [reply discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java]: done

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java (hunk 2 above)
Comment (xingfudeshi) [reply discovery/src/main/java/com/alibaba/fescar/discovery/registry/EurekaRegistryServiceImpl.java]: done,the non-Null check is not necessary here.

**Pull request merged**